File: Modules/ServerTypes/EmployeeProfile/EmployeeResignForm.ts
import { IntegerEditor, DateEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface EmployeeResignForm {
    NoticePeriod: IntegerEditor;
    ResignationDate: DateEditor;
    ResignLeaveDate: DateEditor;
}

export class EmployeeResignForm extends PrefixedContext {
    static formKey = 'EmployeeProfile.EmployeeResign';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!EmployeeResignForm.init)  {
            EmployeeResignForm.init = true;

            var w0 = IntegerEditor;
            var w1 = DateEditor;

            initFormType(EmployeeResignForm, [
                'NoticePeriod', w0,
                'ResignationDate', w1,
                'ResignLeaveDate', w1
            ]);
        }
    }
}

File: Modules/HumanResource/EmployeeProfile/EmployeeResign/EmployeeResignForm.cs
using Serenity.ComponentModel;
using System;

namespace HRMSoftware.EmployeeProfile.Forms;

[FormScript("EmployeeProfile.EmployeeResign")]
[BasedOnRow(typeof(EmployeeResignRow), CheckNames = true)]
public class EmployeeResignForm
{



    [HalfWidth]
    public int NoticePeriod { get; set; }
    [HalfWidth]
    public DateTime ResignationDate { get; set; }
    public DateTime ResignLeaveDate { get; set; }
}